storage/entities/peer.js

```diff
@@ -0,0 +1,307 @@
+	 * @param {Number} [options.offset=0] - Offset to start the records
+	 * @param {fieldSets.Peer} [options.fieldSet='FIELD_SET_SIMPLE'] - Fieldset to choose
+	 * @param {Object} tx - Database transaction object
+	 * @return {Promise.<BasicPeer|ExtendedPeer, Error>}
+	 */
+	getOne(filters, options = {}, tx) {
+		this.validateFilters(filters);
+		this.validateOptions(options);
+		const parsedOptions = _.defaults(
+			{},
+			_.pick(options, ['limit', 'offset', 'fieldSet']),
+			_.pick(this.defaultOptions, ['limit', 'offset', 'fieldSet'])
+		);
+		const mergedFilters = this.mergeFilters(filters);
+		const parsedFilters = this.parseFilters(mergedFilters);
+
+		const params = Object.assign(
+			{},
+			{ limit: parsedOptions.limit, offset: parsedOptions.offset },
+			{
+				parsedFilters,
+			}
+		);
+
+		return this.adapter.executeFile(
+			{
+				[Peer.prototype.FIELD_SET_SIMPLE]: this.SQLs.selectSimple,
+			}[parsedOptions.fieldSet],
+			params,
+			{ expectedResult: 1 },
+			tx
+		);
+	}
+
+	/**
+	 * Get list of peers
+	 *
+	 * @param {filters.Peer|filters.Peer[]} [filters = {}]
+	 * @param {Object} [options = {}] - Options to filter data
+	 * @param {Number} [options.limit=10] - Number of records to fetch
+	 * @param {Number} [options.offset=0] - Offset to start the records
+	 * @param {fieldSets.Peer} [options.fieldSet='FIELD_SET_SIMPLE'] - Fieldset to choose
+	 * @param {Object} tx - Database transaction object
+	 * @return {Promise.<BasicPeer[]|ExtendedPeer[], Error>}
+	 */
+	get(filters = {}, options = {}, tx) {
+		this.validateFilters(filters);
+		this.validateOptions(options);
+		const mergedFilters = this.mergeFilters(filters);
+		const parsedFilters = this.parseFilters(mergedFilters);
+		const parsedOptions = _.defaults(
+			{},
+			_.pick(options, ['limit', 'offset', 'fieldSet']),
+			_.pick(this.defaultOptions, ['limit', 'offset', 'fieldSet'])
+		);
+
+		const params = Object.assign(
+			{},
+			{ limit: parsedOptions.limit, offset: parsedOptions.offset },
+			{ parsedFilters }
+		);
+
+		return this.adapter.executeFile(
+			{
+				[this.FIELD_SET_SIMPLE]: this.SQLs.selectSimple,
+			}[parsedOptions.fieldSet],
+			params,
+			{},
+			tx
+		);
+	}
+
+	/**
+	 * Create peer object
+	 *
+	 * @param {Object} data
+	 * @param {Object} [options]
+	 * @param {Object} tx - Transaction object
+	 * @return {*}
+	 */
+	// eslint-disable-next-line no-unused-vars
+	create(data, options = {}, tx) {
+		const objectData = _.defaults(data, defaultCreateValues);
+		const createSet = this.getValuesSet(objectData);
+		const attributes = Object.keys(data)
+			.map(k => `"${this.fields[k].fieldName}"`)
+			.join(',');
+
+		return this.adapter.executeFile(
+			this.SQLs.create,
+			{ createSet, attributes },
+			{},
+			tx
+		);
+	}
+
+	/**
+	 * Update the records based on given condition
+	 *
+	 * @param {filters.Peer} [filters]
+	 * @param {Object} data
+	 * @param {Object} [options]
+	 * @param {Object} tx - Transaction object
+	 * @return {*}
```

Comment:
?

Reply:
The adapter returns an empty array we may want to change this behaviour in the branch making general changes? @nazarhussain opinions?

Reply:
After discussing with Nazar this was changed to expect no rows and return null for udpate and create.

Reply:
Done

Reply:
Wouldn't it be better to return `boolean` as feedback for the consumer?

Reply:
@lsilvs the way the current implementation works is that it returns an array, null or one and assumes that if there was no error thrown then the operation was successful. Should we move this discussion to the PR where general topics are being refactored? 
@nazarhussain opinions about what to return?